### DISCUSSION TITLE
Relationship Picker gives multi options for a single selection

### DISCUSSION THREAD

**Discussion by amirshabanics** [Budibase/budibase#5916] [Help] [answered]
Hi,
I tried to create a page for creating a model in Budibase. The model has a field with a foreign key to another model.

This is my model in Prisma for better understanding the schema:
``` prisma
model CommunityListItem {
  community        Community      @relation(fields: [communityId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  communityId      String
}
model Community {
  id                         String        @id @default(cuid())
  CommunityListItem         CommunityListItem[]
}
```

I fetched the database and defined all relationships I needed like this:
![image](https://user-images.githubusercontent.com/46162834/168650528-d50131d6-2289-481f-ab5c-60e05dc37f1e.png)

But when I want to add the `Community` field to the group field, insert a form field with multi-selection.
![image](https://user-images.githubusercontent.com/46162834/168650896-3953218e-50d6-4d82-b781-d8a24ff72e5f.png)
 It is not correct and must be a single selection. I don't understand why Budibase acts like that. This situation happened to me when I was trying to fetch tables from the database to add new tables.

Thanks for helping me, and sorry for my bad writing.

**Comment by melohagan** [comment, marked as answer]
Hi @amirshabanics 

Could you show me your **Define Relationships** modal please. E.g.
<img width="358" alt="Screenshot 2022-05-17 at 14 18 22" src="https://user-images.githubusercontent.com/101575380/168820208-f310a097-e8f9-47ba-bdab-e08cacc5cf55.png">

Defined relationships can be either One -> Many, or Many -> Many

It looks like maybe the *Community* -> *CommunityListItem* relationship is defined back to front.

**Reply by amirshabanics** [reply]
![image](https://user-images.githubusercontent.com/46162834/169258061-e6e91004-ee79-4b14-b464-bebd806d7e01.png)

sure, I checked it before but it's not the problem.

**Reply by melohagan** [reply]
Are you definitely using a **Relationship Picker** and not a **Multi-select** ?

![Screenshot 2022-05-18 at 16 22 57](https://user-images.githubusercontent.com/101575380/169079665-7b2af227-fc87-484d-b73c-1828fccd6dc9.png)

**Reply by amirshabanics** [reply]
Yes. I don't know why! But when I try using a **Relationship Picker** for **Community**, it's a multi-selector. I have another field that defines one to may relationship between them and work correctly. (**ExplorePageRow** in the first picture)

**Reply by melohagan** [reply]
Very strange... would you be able to share the app export? I would need the create scripts for the community tables too please.

**Reply by amirshabanics** [reply]
Yes, I know. That sound weird, but I directly changed the app export and fixed it. just change **many-to-one** to **one-to-many** in relationship of **Community** field in  **CommunityListItem** table.

Thanks for helping me.
